### PR TITLE
Add Webpack

### PR DESCRIPTION
**Pull request by lebris**
Based on @skimi work.

### CODE REVIEW

File: src/Webpack/Manifest.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Onyx\Webpack;
+
+class Manifest
+{
+    private
+        $files,
+        $chunkManifest = '';
```
Comment (Niktux) [inline src/Webpack/Manifest.php]: inlined init

File: src/Webpack/Manifest.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    public function getFiles(): array
+    {
+        return $this->files;
+    }
+
+    private function setFiles(string $manifestPath): void
+    {
+        $manifest = json_decode($this->loadManifest($manifestPath), true);
+
+        if (!$manifest)
```
Comment (Niktux) [inline src/Webpack/Manifest.php]: CS sur tous les if

File: src/Webpack/TwigExtension.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+     * Outputs list of files from the webpack compilation manifest
+     * @param  $string $includePattern A glob expression of files to include
+     * @param  $string $excludePattern A glob expression of files to exclude
+     * @return Array                   List of file paths
+     */
+    public function webpackAssets(?string $includePattern = null, ?string $excludePattern = null): array
+    {
+        $validFiles = $this->manifest->getFiles();
+
+        if (!empty($includePattern))
```
Comment (Niktux) [inline src/Webpack/TwigExtension.php]: CS

File: src/Webpack/TwigExtension.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            $validFiles = $this->filterAssets($includePattern);
+        }
+
+        if (!empty($excludePattern))
```
Comment (Niktux) [inline src/Webpack/TwigExtension.php]: CS

File: src/Webpack/TwigExtension.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        return $validFiles;
+    }
+
+    private function filterAssets(string $includeGlob): array
+    {
+        $assets = $this->manifest->getFiles();
+        $filteredAssets = [];
+        $includePatterns = explode(',', $includeGlob);
+
+        foreach ($includePatterns as $globPattern)
```
Comment (Niktux) [inline src/Webpack/TwigExtension.php]: CS

File: src/Webpack/TwigExtension.php [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * Returns the path for a single file
+     * @param  string $name Exact name of the file
+     * @return string       Path to the file
+     */
+    public function webpackAsset(string $name): ?string
+    {
+        $files = $this->manifest->getFiles();
+
+        if (!isset($files[$name]))
```
Comment (Niktux) [inline src/Webpack/TwigExtension.php]: CS

File: src/Webpack/Manifest.php (hunk 1 above)
Comment (lebris) [inline src/Webpack/Manifest.php]: Fix 7be3690

File: src/Webpack/Manifest.php (hunk 2 above)
Comment (lebris) [inline src/Webpack/Manifest.php]: Fix 67f2230

File: src/Webpack/TwigExtension.php (hunk 3 above)
Comment (lebris) [inline src/Webpack/TwigExtension.php]: Fix 67f2230

File: src/Webpack/TwigExtension.php (hunk 4 above)
Comment (lebris) [inline src/Webpack/TwigExtension.php]: Fix 67f2230

File: src/Webpack/TwigExtension.php (hunk 5 above)
Comment (lebris) [inline src/Webpack/TwigExtension.php]: Fix 67f2230

File: src/Webpack/TwigExtension.php (hunk 6 above)
Comment (lebris) [inline src/Webpack/TwigExtension.php]: Fix 67f2230

**Pull request merged**